new documentation build warning

From #28779 which included a closing `@}` in the header without any opening one.  Probably some of the macros in this are supposed to be in user documentation

```
 ---- New errors/warnings not tracked as .known-issues/, please fix them ----

/mnt/nordic/zp/zephyr/include/sys/util_macro.h:577: warning: unbalanced grouping commands
```

--- a/include/sys/util_macro.h
+++ b/include/sys/util_macro.h
@@ -21,6 +21,11 @@
 extern "C" {
 #endif
 
+/**
+ * @addtogroup sys-util
+ * @{
+ */
+
 /*
  * Most of the eldritch implementation details for all the macrobatics
  * below (APIs like IS_ENABLED(), COND_CODE_1(), etc.) are hidden away

